src/Agent.Listener/WorkerManager.cs

```diff
@@ -41,54 +42,73 @@ public void Dispose()
             }
         }
              
-        private ConcurrentDictionary<Guid, JobDispatcherItem> _jobsInProgress 
-            = new ConcurrentDictionary<Guid, JobDispatcherItem>();
+        private Dictionary<Guid, JobDispatcherItem> _jobsInProgress 
+            = new Dictionary<Guid, JobDispatcherItem>();
+        private bool _disposed;
+        private readonly object _lock = new Object();
```

Comment:
my understanding is that when using async/await we may end up running in different threads. For example the code that does postJobDispatcherAction "_jobsInProgress.Remove(jobRequestMessage.JobId);" can run in a different thread than WorkerManager.Dispose.